### PR TITLE
Add naughty.destroy_all_notifications functions.

### PR DESCRIPTION
**Pull request by morethanoneanimal**
Update connect_signal in naughty lib to use destroy_all_notifications().

I have this function in my config and it comes in handy. I think it's a good candidate to an API.

### CODE REVIEW

**Comment by Elv13** [comment]
Hello, thanks for this, this is a little issue to fix:

     lib/naughty/core.lua:375:12: shadowing upvalue screen on line 21

File: lib/naughty/core.lua [hunk 1]
```diff
@@ -364,6 +360,27 @@ function naughty.destroy(notification, reason, keep_visible)
     end
 end
 
+--- Destroy all notifications on given screens.
+--
+-- @param screens Screens on which notifications should be destroyed. If nil,
+-- destroy notifications on all screens.
+-- @param reason One of the reasons from notificationClosedReason.
```
Comment (blueyed) [inline lib/naughty/core.lua]: Please use `@tparam`.

Comment (morethanoneanimal) [inline lib/naughty/core.lua]: Done. Also, I added return value to be consistent with naughty.destroy function.

**Comment by morethanoneanimal** [comment]
Fixed var shadowing issue.

**Comment by blueyed** [comment]
Thanks!

**Pull request merged**